server/src/main/java/org/jboss/as/server/ApplicationServerService.java

```diff
@@ -200,6 +203,21 @@ public synchronized void stop(final StopContext context) {
         CurrentServiceContainer.setServiceContainer(null);
         String prettyVersion = configuration.getServerEnvironment().getProductConfig().getPrettyVersionString();
         ServerLogger.AS_ROOT_LOGGER.serverStopped(prettyVersion, Integer.valueOf((int) (context.getElapsedTime() / 1000000L)));
+        deleteStartupMarker();
+    }
+
+    private void deleteStartupMarker() {
+        String tmpDir = System.getProperty("jboss.server.temp.dir");
```

Comment:
This is available from the ServerEnvironment, which is available from the 'configuration' field.